MOPTools/MOP_Literature_Extraction/upload.py

```diff
@@ -0,0 +1,1336 @@
+                                                                    hasStepDuration=duration,
+                                                                    hasVessel=vessel, 
+                                                                    hasAddedChemicalInput=chemical_input, 
+                                                                    isStirred=standard_step['stir'], 
+                                                                    hasTargetPh=float(standard_step['targetPH']), 
+                                                                    isLayered=standard_step["isLayered"], 
+                                                                    hasVesselEnvironment=atmosphere,
+                                                                    rdfs_comment=standard_step['comment'])  
+        # Push the components to the KG
+        components = [add_class, vessel, duration, duration_value]
+        uputil.push_component_to_kg(components, synthesis_client)
+        # Update the chemicals list
+        chemicals_list.append(chemical_input)
+        return add_class, vessel_list, chemicals_list
+    # Case: HeatChill step
+    elif "HeatChill" in standard_input:
+        standard_step                                           = standard_input["HeatChill"]
+        # Preprocess step to extract duration, vessel, and atmosphere
+        vessel, vessel_list, duration, duration_value, atmosphere, id_hash_value         = uputil.steps_preupload(standard_step, synthesis_client, vessel_list)
+        # Extract target temperature
+        if standard_step["targetTemperature"] == "room temperature":
+            temp                                    = [25.0]                # Default room temperature
+            temperature_unit                        = uputil.get_unit("C", synthesis_client) 
+        else:
+            temp, temp_unit                         = uputil.extract_numbers_and_units(standard_step["targetTemperature"],"temp")
+            temperature_unit                        = uputil.get_unit(temp_unit[0], synthesis_client) 
+            print("temperature: ", temp, temp_unit)
+        # Extract heating/cooling rate
+        heat_rate, rate_unit                        = uputil.extract_numbers_and_units(standard_step["heatingCoolingRate"], "temp")
+        rate_unit                                   = uputil.get_unit(rate_unit[0], synthesis_client) 
+        device                                      = HeatChillDevice(rdfs_label=standard_step["usedDevice"])
+        print("heatingrate: ", heat_rate, rate_unit)
+        # temperature 
+        id_hash_value                               = str(uuid.uuid4())
+        # Instantiate temperature and rate measures
+        temperature_value                           = Measure(
+            instance_iri=f"https://www.theworldavatar.com/kg/OntoSyn/TemperatureValue_{id_hash_value}",
```

Comment:
Again, hard-coding such strings is problematic. At least the base IRI should be a string constant defined somewhere. If it is not practical to fix this quickly across the whole code, then you may want to post this as an issue.